src/Transactions/TransactionFields/LinesField.php

```diff
@@ -12,16 +12,32 @@
 trait LinesField
 {
     /**
-     * @var TransactionLine[]
+     * @var TransactionLine[][]
      */
-    private $lines = [];
+    private $lines_per_type = [];
```

Comment:
If you make the `LineType` constants public, you can initialise this array with empty lists. It would not only make the structure of the contents of this variable clear, it would also allow you to remove all `array_key_exists()` checks in this trait.

```php
private $lines_per_type = [
    LineType::TOTAL  => [],
    LineType::DETAIL => [],
    LineType::VAT    => [],
];

Reply:
Good point! I wanted to do this but couldn't because I couldn't actually call `LineType::TOTAL()`, but it can be done like this. I guess this is one of the reasons to prefer having the constants of the enums publicly available.

Reply:
This has really simplified the class, being able to lose the additional methods as well, nice :)